kube/Cargo.toml

```diff
@@ -49,7 +49,7 @@ either = { version = "1.6.1", optional = true }
 thiserror = "1.0.23"
 futures = { version = "0.3.8", optional = true }
 pem = { version = "0.8.2", optional = true }
-openssl = { version = "0.10.32", optional = true }
+openssl = { version = "0.10.32", optional = true, features = ["vendored"] }
```

Comment:
`vendored` is definitely easier to build, but some users might not want this.

An argument against vendored static OpenSSL is the binary ending up with outdated OpenSSL in the future. We're using `openssl` only to make PKCS12 archive (with more private key types supported than pure Rust solutions), so I'm not sure how significant that can be.

Reply:
I forgot that `native-tls` uses `openssl` on Linux. Is enabling `openssl/vendored` enough?

Reply:
Found https://doc.rust-lang.org/cargo/reference/resolver.html#feature-resolver-version-2

I guess this will only enable `vendored` on Windows?
```toml
[dependency.openssl]
version = "0.10.32"
optional = true

[target.'cfg(windows)'.dependencies.openssl]
version = "0.10.32"
features = ["vendored"]
```

Need
```toml
[workspace]
resolver = "2"
```
https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions

Reply:
ah, that's a good solution, i'm +1 on that.

Reply:
hmm the resolver change does seem to impact k8s-openapi feature selection somehow. At least build failed on that. Will have to look at it tomorrow :-(

Reply:
> Features enabled on dev-dependencies will not be unified when those same dependencies are used as a normal dependency, unless those dev-dependencies are currently being built.

So `cargo build` fails. Skipping that step should be fine, right (it's built when testing)? I just hope this doesn't cause errors when editing.

Reply:
i'm sure we can make it work, yeah, but i'd rather not make the entire class of `cargo build` commands fail:

- cargo build
- cargo build -p kube
- cargo build -p kube-core
- cargo build -p kube-runtime

i feel this is a pretty steep price to pay for this. I don't think the resolver is well-suited to our use-case.

I would personally rather re-export a native-tls-vendored feature for this.

Reply:
tried to add some caching via https://github.com/marketplace/actions/run-vcpkg but when i thought i got it working, it gave me a [link error](https://github.com/clux/kube-rs/runs/2903061264?check_suite_focus=true).

don't think I have it in me to sort windows out right now, but also don't want to swath enable `vendored` (which is what happens without resolver2) because it represents a pretty big increase in build times locally for me (30% real time, 90% user time).

am considering just commenting out the windows build temporarily and opening another issue for this, this is clearly non-trivial.

Reply:
Opened #566 with the results of my experimentation. Hopefully it makes it easier for the next person when they have time.